### DISCUSSION TITLE
Sleep timer

### DISCUSSION THREAD

**Discussion by Vs1012007** [MissingCore/Music#106] [Ideas]
It would be great if you could add sleep timer to the music app

**Comment by cyanChill** [comment]
I think this is an interesting idea, though I have some concerns and some questions about expected behavior. I guess we'll start with the questions: What behavior do you expect from this feature? Should we stop after the current track ends? Should we stop after X minutes? Should we stop at a certain time?

Some concerns I have include:
- The feature not working as intended (ie: it doesn't stop at the specified time). This might happen as different behaviors might occur across different Android versions and devices (ie: something used to implement this feature might break).
- Kind of related to the previous point, is battery consumption if the feature doesn't work (as music will continue playing until you wake up and turn it off or when the battery dies).

If I ever do get to implementing this feature, it'll probably be marked as experimental for these reasons.

**Comment by cyanChill** [comment]
As of [`v2.4.0-rc.1`](https://github.com/MissingCore/Music/releases/tag/v2.4.0-rc.1) this is available as an **experimental feature**. What this means is that you need to enable the feature in the settings, for which the UI for this will then be available.

When the experiment is enabled, a new button will appear on the "Now Playing" screen allowing you to create a sleep timer.
